rust/connlib/tunnel/src/client.rs

```diff
@@ -1024,8 +1063,14 @@ impl IpProvider {
 
 #[cfg(test)]
 mod tests {
+    use rand_core::OsRng;
+
     use super::*;
 
+    fn client_state_fixture() -> ClientState {
+        ClientState::new(StaticSecret::random_from_rng(OsRng))
+    }
```

Comment:
What I did for `Allocation` is make a `for_test` ctor using an `impl` block scoped to the `tests` module. I've found that reads better in the test because it is "just another ctor".